vignettes/markdown.Rmd

```diff
@@ -1,17 +1,188 @@
-<!--
-%\VignetteEngine{knitr::knitr}
-%\VignetteIndexEntry{Write R documentation in Markdown}
--->
+---
+title: "Write R Documentation in Markdown"
+author: "Gábor Csárdi"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Write R Documentation in Markdown}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
 
-```{r, echo = FALSE, message = FALSE}
-knitr::opts_chunk$set(
-  comment = "#>",
-  error = FALSE,
-  tidy = FALSE
-)
+```{r, include = FALSE}
+knitr::opts_chunk$set(comment = "#>", collapse = TRUE)
 ```
 
-# Write R documentation in Markdown
+# Introduction
 
-This is a vignette to be written once we know what exactly the Markdown
-support will cover.
+Starting from version 6.0.0, roxygen supports markdown markup within most roxygen tags. Roxygen uses the `commonmark` package, which is based on the CommonMark Reference Implementation to parse these tags. See http://commonmark.org/help/ for more about the parser and the markdown language it supports.
+
+# Turning on markdown support
+
+Currently you have to turn on markdown formatting manually for each roxygen chunk. You can do this by using the `@md` tag anywhere in the chunk. In the future roxygen will provide a global switch that turns on markdown formatting for the whole package. (And also a `@noMd` to turn it off, optionally.)
+
+Here is an example roxygen chunk that uses markdown.
+
+```r
+#' Use roxygen to document a package.
+#'
+#' This function is a wrapper for the [](roxygen2::roxygenize) function from 
```

Comment:
How about bringing back the backtick notation, e.g. whenever the backticks contain a single instance of `::`, it is picked up as a link to a function. E.g.:

```
The function `roxygen2::roygenize`.
Or just simply `::roxygenize`.
```

We would also keep the current forms, to have a consistent notation, and the backtick forms would be just shortcuts. 

Another alternative would be to use the autolink form as a shortcut, but this will not allow the `::` qualified form, at least not always:

```
This one is fine: <roxygenize>.
This one, too: <pkg::func>. 
But this one is not: <roxygen2::roxyigenize>. Probably because of the number in the tag?
```

I'll dig a bit more.....